openupgradelib/openupgrade_merge_records.py

```diff
@@ -431,6 +431,111 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
         )
 
 
+def _adjust_merged_values_sql(env, model_name, record_ids, target_record_id,
+                              exclude_columns, model_table, field_spec):
+    """This method deals with the values on the records to be merged +
+    the target record, performing operations that makes sense on the meaning
+    of the model.
+
+    :param field_spec: Dictionary with field names as keys and forced operation
+      to perform as values. If a field is not present here, default operation
+      will be performed.
+
+      Possible operations by field types same as _adjust_merged_values_orm.
+    """
+    if not column_exists(env.cr, model_table, 'id'):
+        return
+    env.cr.execute(
+        """
+        SELECT isc.column_name, isc.data_type, imf.ttype
```

Comment:
You should get fields from `ir_model_fields`, not from here, as the existing columns may not be the real fields one (you can have more and perform operations in then incorrectly). The contrary may happen the same (non stored fields, m2m/o2m fields), but you should intersect both for having the proper one, and add the TODO to handle here m2m and o2m fields.

Reply:
But there is a JOIN with ir_model_fields two lines below 🤔

Reply:
OK, right, then please add the TODO for the m2m/o2m.